frontend/src/lib/utils.ts

```diff
@@ -100,23 +88,18 @@ export function requireNonNull<T>(obj: T | null | undefined): T {
 }
 
 /**
- * Provide an ImmutableSet of SimpleElements by walking a BlockElement to
- * its leaves.
+ * A type predicate that is true if the given value is not undefined.
  */
-export function flattenElements(
-  elements: BlockElement
-): ImmutableSet<SimpleElement> {
-  return elements.reduce(
-    (flattened: ImmutableSet<SimpleElement>, reportElement: ReportElement) => {
-      const element = reportElement.get("element")
-
-      if (element instanceof List) {
-        return flattened.union(flattenElements(element as BlockElement))
-      }
-      return flattened.union(ImmutableSet.of(element as SimpleElement))
-    },
-    ImmutableSet.of<SimpleElement>()
-  )
+export function notUndefined<T>(value: T | undefined): value is T {
+  return value !== undefined
+}
+
+/**
+ * A type predicate that is true if the given value is neither undefined
+ * nor null.
+ */
+export function notNull<T>(value: T | null | undefined): value is T {
```

Comment:
It's used in `Block.renderElements` to filter non-null elements out of a list. It's a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types) (as opposed to a simple boolean predicate) because this allows TypeScript to deduce that the resulting array type is `ReactNode[]` as opposed to `(ReactNode | null)[]`.

I'm happy to move it into `Block.tsx` and make it private, but it seems like a pretty innocuous function to me!

Reply:
Oh, I just did a ctrl+f for "notNull" but Block.tsx was collapsed in Github's editor. Makes sense to have it.